nullaway/src/main/java/com/uber/nullaway/generics/PreservedAnnotationTreeVisitor.java

```diff
@@ -156,6 +160,32 @@ private static MethodHandle createHandle() {
       }
     }
 
+    private static MethodHandle createGetMetadataHandle() {
```

Comment:
Remind me, all of these methods with reflection trickery are to avoid having a compile-time dep on JDK21, correct?

Reply:
A bunch of (internal) javac APIs changed between JDK 17 and 21 related to these `TypeMetadata` objects.  On JDK 21+ we use reflection to operate on them since I expect that is used less for compiling than JDK 17 and before.  Unfortunately I think we can only drop this reflection approach once we drop support for building on JDK 17.  This is hacky, but the alternative that doesn't rely on internal APIs would require building up our own type data structures from scratch, like Checker Framework, which would be a lot more code.